frontend/lib/src/hooks/useSubmitFormViaEnterKey.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback } from "react"
+
+import { WidgetStateManager } from "@streamlit/lib/src/WidgetStateManager"
+
+const isEnterKeyPressed = (
+  event: Partial<React.KeyboardEvent<HTMLElement>> & {
+    metaKey: boolean
+    ctrlKey: boolean
+    /** @deprecated */
+    keyCode: number
+    key: string
+  }
+): boolean => {
```

Comment:
**suggestion:** It seems like this type is to help limit to only necessary props. I think `pick` is a better option here:

```suggestion
const isEnterKeyPressed = (
  event: Pick<
    React.KeyboardEvent<HTMLElement>,
    "metaKey" | "ctrlKey" | "keyCode" | "key" | "nativeEvent"
  >
): boolean => {
```

Reply:
Yeah great point! I thought about this and one a little bit annoying issue is that `Pick` makes the picked properties required, even if they originally were optional. There is a way to persist this optionality by merging different definitions, but then it becomes pretty unreadable. This point is not too bad though, we only have to update the unit tests to include the fields in the passed object. If you prefer that, I will make the update 👍

Reply:
You bring up a great point about optionality aspect of `Pick`. I would definitely prefer us to not redeclare types. I tried this locally and this should be enough to work:

```tsx
type SubmitFormKeyboardEvent = Pick<
  React.KeyboardEvent<HTMLElement>,
  "metaKey" | "ctrlKey" | "keyCode" | "key" | "nativeEvent" | "preventDefault"
>
```

And in tests we can simply sidestep the fields that aren't being utilized:

```tsx
    onInputChange.current({
      /* existing args */
      nativeEvent: undefined as never,
      preventDefault: vi.fn(),
    })
```

Reply:
when you say merging different definitions, do you mean something like this?
`type NewType = Pick<Original, 'required' | 'properties'> & Partial<Pick<Original, 'optional' | 'properties'>>;`

with how long typescript has been around, kinda disappointed there isn't a cleaner way to achieve this :(

Reply:
@sfc-gh-bnisco Great, I will update it!

@sfc-gh-nbellante Yeah exactly, something like that! And I agree, I had the same feeling :(